### PR TITLE
Add ZipArchive and UnzipArchive tasks

### PR DESCRIPTION
**Pull request by natemcmaster**
Add tasks to unzip and zip archive files
Add unit tests for tasks

Part of work for updating the libuv build https://github.com/aspnet/libuv-package/issues/16

### CODE REVIEW

**Comment by moozzyk** [comment]
LOL https://github.com/moozzyk/MSBuild-Tasks/blob/master/common.tasks#L4 

Why is it needed for libuv-package?

**Comment by natemcmaster** [comment]
Working on simplifying libuv-package. Instead of reading a project.lock.json file and finding files in a nuget cache, we can simple extract a zip files. Also, we may be able to reduce down to one repository instead of 2.

File: src/Internal.AspNetCore.BuildTools.Tasks/UnzipArchive.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    /// Unzips an archive file.
+    /// </summary>
+    public class UnzipArchive : Task
+    {
+        /// <summary>
+        /// The path where the zip file should be created.
+        /// </summary>
+        [Required]
+        public string File { get; set; }
+
```
Comment (moozzyk) [inline src/Internal.AspNetCore.BuildTools.Tasks/UnzipArchive.cs]: nit: other properties have doc comments

**Comment by natemcmaster** [comment]
cref https://github.com/Microsoft/msbuild/issues/1781

File: src/Internal.AspNetCore.BuildTools.Tasks/UnzipArchive.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+namespace Microsoft.AspNetCore.BuildTools
+{
+    /// <summary>
+    /// Unzips an archive file.
+    /// </summary>
+    public class UnzipArchive : Task
+    {
+        /// <summary>
+        /// The path where the zip file should be created.
```
Comment (moozzyk) [inline src/Internal.AspNetCore.BuildTools.Tasks/UnzipArchive.cs]: Feels like this comment is for zipping the file

File: src/Internal.AspNetCore.BuildTools.Tasks/ZipArchive.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        public override bool Execute()
+        {
+            var workDir = FileHelpers.EnsureTrailingSlash(WorkingDirectory);
+
+            foreach (var file in SourceFiles)
+            {
+                if (!string.IsNullOrEmpty(file.GetMetadata("Link")))
+                {
+                    continue;
+
```
Comment (moozzyk) [inline src/Internal.AspNetCore.BuildTools.Tasks/ZipArchive.cs]: nit: empty line

File: src/Internal.AspNetCore.BuildTools.Tasks/ZipArchive.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                if (!file.ItemSpec.StartsWith(workDir))
+                {
+                    Log.LogError("Item {0} is not inside the working directory {1}. Set the metadata 'Link' to file path that should be used within the zip archive",
+                        file.ItemSpec,
+                        workDir);
+                    return false;
+                }
+
+                file.SetMetadata("Link", file.ItemSpec.Replace(workDir, ""));
```
Comment (moozzyk) [inline src/Internal.AspNetCore.BuildTools.Tasks/ZipArchive.cs]: Wouldn't this break for a case where link is `temp\temp\myFile.txt` and the `WorkingDirectory` is `temp`?

**Comment by natemcmaster** [comment]
:up: :date: pr feedback + add more tests

**Pull request merged**